gcp-observability/src/main/java/io/grpc/gcp/observability/ObservabilityConfigImpl.java

```diff
@@ -100,19 +102,19 @@ private void parseConfig(Map<String, ?> config) {
         }
         this.eventTypes = eventTypesBuilder.build();
       }
-      String sampler = JsonUtil.getString(config, "global_trace_sampler");
       Double samplingRate = JsonUtil.getNumberAsDouble(config, "global_trace_sampling_rate");
-      checkArgument(
-          sampler == null || samplingRate == null,
-          "only one of 'global_trace_sampler' or 'global_trace_sampling_rate' can be specified");
-      if (sampler != null) {
-        this.sampler = new Sampler(SamplerType.valueOf(sampler.toUpperCase()));
+      if (enableCloudTracing && samplingRate == null) {
+        this.sampler = Samplers.probabilitySampler(0.0);
       }
       if (samplingRate != null) {
         checkArgument(
             samplingRate >= 0.0 && samplingRate <= 1.0,
-            "'global_trace_sampling_rate' needs to be between 0.0 and 1.0");
-        this.sampler = new Sampler(samplingRate);
+            "'global_trace_sampling_rate' needs to be between [0.0, 1.0]");
+        if (samplingRate == 1.0) {
```

Comment:
Comparison of double numbers is tricky and because of precision (and conversion from String to double) issues this equality check may not go through. Has this been unit tested? https://www.baeldung.com/java-comparing-doubles#comparing_in_plain_Java

Reply:
Updated. PTAL